### ISSUE TITLE
SELinux volume permission issues when playing kube file (rootless)

### ISSUE THREAD

**Issue by Ian2020** [containers/podman#10938]
**Description**

This is a new issue but based on [SELinux volume permission issues when playing kube file (rootless) · Issue #9371 · containers/podman](https://github.com/containers/podman/issues/9371) which does not appear to be fixed.

As noted in that bug the fix seems to be available, just not in the 3.2 branch, see this [comment](https://github.com/containers/podman/issues/9371).

**Steps to reproduce the issue:**

SELinux is set to "enforcing".

Scenario A: running everything manually:

1. `podman pod create -n testpod`
2. create a simple Dockerfile that writes into a file on a mounted volume:

```bash
cat > Dockerfile << "EOF"
FROM docker.io/centos:centos8
USER root
CMD while true; do echo $(date) >> /testdir/testfile && sleep 3 ; done
EOF
```

3. build it `podman build .`, create a testdir `mkdir testdir` then run the container in the pod: `podman run -d --rm --name testcontainer --pod testpod -v $(pwd)/testdir:/testdir:Z testimage`
4. let's look at the testfile:

```bash
 tail $(pwd)/testdir/testfile
Mon Feb 15 13:04:13 UTC 2021
Mon Feb 15 13:04:16 UTC 2021
Mon Feb 15 13:04:19 UTC 2021
Mon Feb 15 13:04:22 UTC 2021
Mon Feb 15 13:04:25 UTC 2021
Mon Feb 15 13:04:28 UTC 2021
```

So far so good.

5. Tidy up:

```bash
podman pod stop testpod
podman pod rm testpod
rm -rf $(pwd)/testdir
```

Scenario B: using podman generate:

1. `podman pod create -n testpod`
2. create the same simple Dockerfile:

```bash
cat > Dockerfile << "EOF"
FROM docker.io/centos:centos8
USER root
CMD while true; do echo $(date) >> /testdir/testfile && sleep 3 ; done
EOF
```

3. build it `podman build .`, create a testdir `mkdir testdir` then run the container in the pod: `podman run -d --rm --name testcontainer --pod testpod -v $(pwd)/testdir:/testdir:Z testimage`
4. generate a YAML file while the container is running: `podman generate kube testpod > kube.yaml`
5. the kube.yaml file looks like this:

<details>

```YAML
# Generation of Kubernetes YAML is still under development!
#
# Save the output of this file and use kubectl create -f to import
# it into Kubernetes.
#
# Created with podman-3.2.2
apiVersion: v1
kind: Pod
metadata:
  creationTimestamp: "2021-07-15T09:30:04Z"
  labels:
    app: testpod
  name: testpod
spec:
  containers:
  - command:
    - /bin/sh
    - -c
    - while true; do echo $(date) >> /testdir/testfile && sleep 3 ; done
    env:
    - name: PATH
      value: /usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
    - name: TERM
      value: xterm
    - name: container
      value: podman
    image: localhost/testimage:latest
    name: testcontainer
    resources: {}
    securityContext:
      allowPrivilegeEscalation: true
      capabilities:
        drop:
        - CAP_MKNOD
        - CAP_NET_RAW
        - CAP_AUDIT_WRITE
      privileged: false
      readOnlyRootFilesystem: false
      runAsGroup: 0
      runAsUser: 0
      seLinuxOptions: {}
    volumeMounts:
    - mountPath: /testdir
      name: home-user-Downloads-testdir-host-0
    workingDir: /
  dnsConfig: {}
  restartPolicy: Never
  volumes:
  - hostPath:
      path: /home/user/Downloads/testdir
      type: Directory
    name: home-user-Downloads-testdir-host-0
status: {}
```

</details>

6. Tidy up:

```bash
podman pod stop testpod
podman pod rm testpod
rm -rf $(pwd)/testdir
```

7. Recreate testdir `mkdir $(pwd)/testdir` and play it: `podman play kube kube.yaml`
8. Check the testfile:

```bash
tail -f $(pwd)/testdir/testfile
tail: cannot open '/home/podman/testdir/testfile' for reading: No such file or directory
```

9. Let's see the logs:

```bash
podman logs testpod-testcontainer
/bin/sh: /testdir/testfile: Permission denied
/bin/sh: /testdir/testfile: Permission denied
/bin/sh: /testdir/testfile: Permission denied
/bin/sh: /testdir/testfile: Permission denied
/bin/sh: /testdir/testfile: Permission denied
```

Scenario B works when SELinux is set to pemissive mode.

**Describe the results you received:**

I would expect the container to write into the testfile in both Scenario A and B also with SELinux set to enforcing mode.

**Describe the results you expected:**

I get "permission denied" error message.

**Additional information you deem important (e.g. issue happens only occasionally):**

I have tried adding `:z` and `:Z` to the mountpath in kube.yaml but they are interpreted as part of the path and otherwise ignored.

**Output of podman version:**

<details>

```bash
podman version
Version:      3.2.2
API Version:  3.2.2
Go Version:   go1.16.4
Built:        Sat Jun 26 00:49:19 2021
OS/Arch:      linux/amd64
```

</details>

**Output of podman info --debug:**

<details>

```bash
host:
  arch: amd64
  buildahVersion: 1.21.0
  cgroupControllers: []
  cgroupManager: systemd
  cgroupVersion: v2
  conmon:
    package: conmon-2.0.27-2.fc34.x86_64
    path: /usr/bin/conmon
    version: 'conmon version 2.0.27, commit: '
  cpus: 4
  distribution:
    distribution: fedora
    version: "34"
  eventLogger: journald
  hostname: purismv4
  idMappings:
    gidmap:
    - container_id: 0
      host_id: 1001
      size: 1
    - container_id: 1
      host_id: 165536
      size: 65536
    uidmap:
    - container_id: 0
      host_id: 1001
      size: 1
    - container_id: 1
      host_id: 165536
      size: 65536
  kernel: 5.12.14-300.fc34.x86_64
  linkmode: dynamic
  memFree: 16759562240
  memTotal: 33576525824
  ociRuntime:
    name: crun
    package: crun-0.20.1-1.fc34.x86_64
    path: /usr/bin/crun
    version: |-
      crun version 0.20.1
      commit: 0d42f1109fd73548f44b01b3e84d04a279e99d2e
      spec: 1.0.0
      +SYSTEMD +SELINUX +APPARMOR +CAP +SECCOMP +EBPF +CRIU +YAJL
  os: linux
  remoteSocket:
    path: /run/user/1001/podman/podman.sock
  security:
    apparmorEnabled: false
    capabilities: CAP_CHOWN,CAP_DAC_OVERRIDE,CAP_FOWNER,CAP_FSETID,CAP_KILL,CAP_NET_BIND_SERVICE,CAP_SETFCAP,CAP_SETGID,CAP_SETPCAP,CAP_SETUID,CAP_SYS_CHROOT
    rootless: true
    seccompEnabled: true
    seccompProfilePath: /usr/share/containers/seccomp.json
    selinuxEnabled: true
  serviceIsRemote: false
  slirp4netns:
    executable: /usr/bin/slirp4netns
    package: slirp4netns-1.1.9-1.fc34.x86_64
    version: |-
      slirp4netns version 1.1.8+dev
      commit: 6dc0186e020232ae1a6fcc1f7afbc3ea02fd3876
      libslirp: 4.4.0
      SLIRP_CONFIG_VERSION_MAX: 3
      libseccomp: 2.5.0
  swapFree: 8589930496
  swapTotal: 8589930496
  uptime: 40h 54m 47.93s (Approximately 1.67 days)
registries:
  search:
  - registry.fedoraproject.org
  - registry.access.redhat.com
  - docker.io
  - quay.io
store:
  configFile: /home/user/.config/containers/storage.conf
  containerStore:
    number: 12
    paused: 0
    running: 2
    stopped: 10
  graphDriverName: overlay
  graphOptions:
    overlay.mount_program:
      Executable: /usr/bin/fuse-overlayfs
      Package: fuse-overlayfs-1.5.0-1.fc34.x86_64
      Version: |-
        fusermount3 version: 3.10.4
        fuse-overlayfs: version 1.5
        FUSE library version 3.10.4
        using FUSE kernel interface version 7.31
  graphRoot: /home/user/.local/share/containers/storage
  graphStatus:
    Backing Filesystem: btrfs
    Native Overlay Diff: "false"
    Supports d_type: "true"
    Using metacopy: "false"
  imageStore:
    number: 34
  runRoot: /run/user/1001/containers
  volumePath: /home/user/.local/share/containers/storage/volumes
version:
  APIVersion: 3.2.2
  Built: 1624664959
  BuiltTime: Sat Jun 26 00:49:19 2021
  GitCommit: ""
  GoVersion: go1.16.4
  OsArch: linux/amd64
  Version: 3.2.2
```

</details>

**Package info (e.g. output of rpm -q podman or apt list podman):**

<details>

```bash
Name         : podman
Epoch        : 3
Version      : 3.2.2
Release      : 1.fc34
Architecture : x86_64
Size         : 46 M
Source       : podman-3.2.2-1.fc34.src.rpm
Repository   : @System
From repo    : updates
Summary      : Manage Pods, Containers and Container Images
```

</details>

**Have you tested with the latest version of Podman and have you checked the Podman Troubleshooting Guide?**

Yes

**Comment by rhatdan** [comment]
Can you give a simple reproducer. I thought I fixed this.

**Comment by Ian2020** [comment]
You did with commit https://github.com/containers/podman/commit/4cc19f9e0f118d5f93a5399d14f56a887926e073, but it's in main branch and not released in 3.2 branch I think?

**Comment by rhatdan** [comment]
Thanks, I could not find it in my quick search.
This will be fixed in podman 3.3 then,  For now you can just set the label to container_file_t on the host.

chcon -t container_file_t -R PATHTOVOL.
